internal/services/netapp/netapp_backup_policy_resource.go

```diff
@@ -0,0 +1,331 @@
+			ValidateFunc: validation.IntBetween(1, 1019),
+		},
+
+		"monthly_backups_to_keep": {
+			Type:         pluginsdk.TypeInt,
+			Default:      1,
+			Optional:     true,
+			ValidateFunc: validation.IntBetween(1, 1019),
+		},
+
+		"enabled": {
+			Type:     pluginsdk.TypeBool,
+			Default:  true,
+			Optional: true,
+		},
+	}
+}
+
+func (r NetAppBackupPolicyResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (r NetAppBackupPolicyResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 90 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.NetApp.BackupPolicyClient
+			subscriptionId := metadata.Client.Account.SubscriptionId
+
+			var model netAppModels.NetAppBackupPolicyModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			id := backuppolicy.NewBackupPolicyID(subscriptionId, model.ResourceGroupName, model.AccountName, model.Name)
+
+			// Validations
+			if errorList := netAppValidate.ValidateNetAppBackupPolicyCombinedRetention(model.DailyBackupsToKeep, model.WeeklyBackupsToKeep, model.MonthlyBackupsToKeep); len(errorList) > 0 {
+				return fmt.Errorf("one or more issues found while performing deeper validations for %s:\n%+v", id, errorList)
+			}
+
+			metadata.Logger.Infof("Import check for %s", id)
+			existing, err := client.BackupPoliciesGet(ctx, id)
+			if err != nil {
+				if !response.WasNotFound(existing.HttpResponse) {
+					return fmt.Errorf("checking for presence of existing %s: %s", id, err)
+				}
+			}
+
+			if !response.WasNotFound(existing.HttpResponse) {
+				return tf.ImportAsExistsError(r.ResourceType(), id.ID())
+			}
+
+			parameters := backuppolicy.BackupPolicy{
+				Location: location.Normalize(model.Location),
+				Tags:     pointer.To(model.Tags),
+				Properties: backuppolicy.BackupPolicyProperties{
+					DailyBackupsToKeep:   pointer.To(model.DailyBackupsToKeep),
+					WeeklyBackupsToKeep:  pointer.To(model.WeeklyBackupsToKeep),
+					MonthlyBackupsToKeep: pointer.To(model.MonthlyBackupsToKeep),
+					Enabled:              pointer.To(model.Enabled),
+				},
+			}
+
+			err = client.BackupPoliciesCreateThenPoll(ctx, id, parameters)
+			if err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+
+			return nil
+		},
+	}
+}
+
+func (r NetAppBackupPolicyResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 120 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.NetApp.BackupPolicyClient
+
+			id, err := backuppolicy.ParseBackupPolicyID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			metadata.Logger.Infof("Decoding state for %s", id)
+			var state netAppModels.NetAppBackupPolicyModel
+			if err := metadata.Decode(&state); err != nil {
+				return err
```

Comment:
```suggestion
				return fmt.Errorf("decoding: %+v", err)
```

Reply:
Resolved all instances in the NetApp provider